veros/core/external/solvers/petsc_.py

```diff
@@ -7,6 +7,35 @@
 from veros.core import utilities
 from veros.core.external.solvers.base import LinearSolver
 from veros.core.operators import numpy as npx, update, update_add, at, flush
+from veros.core.external.poisson_matrix import assemble_poisson_matrix
+
+STREAM_OPTIONS = {
+    "solver_type": "bcgs",
+    "atol": 1e-24,
+    "rtol": 1e-14,
+    "max_it": 1000,
+    "PC_type": "gamg",
+    "pc_gamg_type": "agg",
+    "pc_gamg_reuse_interpolation": True,
+    "pc_gamg_threshold": 1e-4,
+    "pc_gamg_sym_graph": True,
+    "pc_gamg_agg_nsmooths": 2,
+    "mg_levels_pc_type": "jacobi",
```

Comment:
This would make more sense as a "pc_options" object. Something like


```python
STREAM_OPTIONS = {
    ...,
    "pc_type": "gamg",
    "pc_options": {
        "pc_gamg_type": "agg",
        "pc_gamg_reuse_interpolation": True,
        "pc_gamg_threshold": 1e-4,
        "pc_gamg_sym_graph": True,
        "pc_gamg_agg_nsmooths": 2,
        "mg_levels_pc_type": "jacobi",
    }
}
```

Then just iterate over `pc_options` and add them to `petsc_options`.